cockroachdb/templates/job.init.yaml

```diff
@@ -19,8 +19,10 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
   annotations:
-    helm.sh/hook: post-install,post-upgrade
-    helm.sh/hook-delete-policy: before-hook-creation
+    {{- if not .Release.IsInstall }}
+    helm.sh/hook: post-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
```

Comment:
The main issue with this is the job will be left in the cluster and when you do an upgrade, the new image will be patched to this job which is not allowed as job is an immutable resource in kubernetes.

Reply:
Since the `hook-delete-policy` still applies on Upgrade, it removes the old job before creating the one for the upgrade.
Feel free to test it out.